### ISSUE TITLE
Unhelpful error message with circular dependency (part 2)

### ISSUE THREAD

**Issue by timotheecour** [catkin/catkin_tools#430]
related to https://github.com/catkin/catkin_tools/issues/229 but that was closed despite https://github.com/catkin/catkin_tools/issues/229#issuecomment-245881842 showing the fix didn't work (and that was catkin 0.4.2, I'm also seeing the bug with 0.4.3):

```
[build] Found '68' packages in 0.0 seconds.
Traceback (most recent call last):
  File "/usr/bin/catkin", line 9, in <module>
    load_entry_point('catkin-tools==0.4.3', 'console_scripts', 'catkin')()
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 267, in main
    catkin_main(sysargs)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 262, in catkin_main
    sys.exit(args.main(args) or 0)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/cli.py", line 419, in main
    summarize_build=opts.summarize  # Can be True, False, or None
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 305, in build_isolated_workspace
    packages, context, workspace_packages)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 98, in determine_packages_to_be_built
    workspace_package_names = dict([(pkg.name, (path, pkg)) for path, pkg in ordered_packages])
AttributeError: 'str' object has no attribute 'name'
```

possibly also related to https://github.com/catkin/catkin_tools/issues/429 but not sure since it's not hanging

**Comment by timotheecour** [comment]
workaround:

* build catkin_tools from source (https://github.com/catkin/catkin_tools/blob/master/docs/installing.rst#installing-from-source)
* edit brew/lib/python2.7/site-packages/catkin_tools/verbs/catkin_build/build.py:
```
for path, pkg in ordered_packages:
  print(('path, pkg', path, pkg))
```
right above
```
workspace_package_names = dict([(pkg.name, (path, pkg)) for path, pkg in ordered_packages])
```

this'll print suspicious items

**Comment by kopp** [comment]
When you found the right place: Do you think that you can make this a change with pull request?

**Comment by 130s** [comment]
I opened a possible solution https://github.com/ros-infrastructure/catkin_pkg/pull/208

**Comment by jbvo** [comment]
Seems like it's not gonna get merged, instead it's suggested to handle this case in the calling application.

In this case it would need to be checked somewhere around here: https://github.com/catkin/catkin_tools/blob/master/catkin_tools/verbs/catkin_build/build.py#L88 and at least before line 98 of the same file. However I'm not sure at what level of the if statements and for loops the check would need to be done and what further processing should be aborted.

**Comment by 130s** [comment]
2.5 yrs passed, and I hit this again. I have a WIP.

Another bad thing I found is that when this happens the tool doesn't return without raising a flag (I confirmed that with `catkin run_tests` not running any tests due to this error and the entire test passed).

**Comment by mikepurvis** [comment]
Should be fixed by #641.
